skalenetwork/skaled
# Bug/1702 adopt precompiled oracle

fixes #1702 

added two category of test in unit tests
the first one in ClientTest.cpp to test that precompiled contracts return different value depending on current block timestamp
the second one in PrecompiledTests.cpp to test that skaled properly parses historical data in config

tested manually on local machine as following:
- deploy oracle contract
- send oracle request and wait for response
- hard-code response to the attached contract. it will submit the response to `verifyOracleResponse` function
- send txn to the contract that do the `checkOracleResponse` call. see result
[test.txt](https://github.com/skalenetwork/skaled/files/14196821/test.txt)

to verify on qa network:
- deploy oracle contract
- stop one of the skaled containers
- send oracle request and wait for response  (you can use `oracle-demo` repo for it)
- hard-code response to the attached contract. it will submit the response to `verifyOracleResponse` function
- send txn to the contract that do the `checkOracleResponse` call. see result
- do several node rotations on the chain
- turn on the container that was stopped. wait till it catchup the txn with `checkOracleResponse` function and check its status.
- wait for 2 snapshots interval to verify that stateRoot is the same on all nodes
